tests/e2e/performance/csi_tests/test_pvc_multi_snapshot_performance.py

```diff
@@ -1,42 +1,492 @@
-import os
+"""
+Test to run the maximum supportable snapshots
+"""
+
+# Builtin modules
+import datetime
 import logging
-import subprocess
+import tempfile
+import time
 
+# 3ed party modules
+import json
 import pytest
+import yaml
 
-from ocs_ci.ocs import constants, exceptions
-from ocs_ci.ocs.cluster import CephCluster
-from ocs_ci.utility.utils import ocsci_log_path
+# Local modules
 from ocs_ci.framework.testlib import (
     skipif_ocs_version,
     skipif_ocp_version,
-    E2ETest,
     performance,
 )
+from ocs_ci.helpers import helpers
+from ocs_ci.helpers.performance_lib import run_oc_command
+from ocs_ci.ocs import constants, exceptions
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.ocs.ocp import OCP, switch_to_default_rook_cluster_project
+from ocs_ci.ocs.perftests import PASTest
+from ocs_ci.utility import templating
+from ocs_ci.ocs.resources import ocs
+from ocs_ci.framework.testlib import ignore_leftovers
 
 log = logging.getLogger(__name__)
 
-SKIP_REASON = "Skipping multiple snapshot tests until github issue 5197 is addressed"
+# Error message to look in a command output
+ERRMSG = "Error in command"
+# Time formatting in the csi-driver logs
+time_format = "%H:%M:%S.%f"
 
 
 @performance
 @skipif_ocp_version("<4.6")
 @skipif_ocs_version("<4.6")
-class TestPvcMultiSnapshotPerformance(E2ETest):
+class TestPvcMultiSnapshotPerformance(PASTest):
     """
     Tests to measure PVC snapshots creation performance & scale
-    The test is trying to to take the maximum number of snapshot for one PVC
+    The test is trying to to take the maximal number of snapshot for one PVC
     """
 
-    @pytest.mark.skip(SKIP_REASON)
+    def setup(self):
+        """
+        Setting up the test environment :
+            Calculating the amount of storage which available for the test
+            Creating namespace (project) for the test
+
+        """
+        log.info("Setting up the test environment")
+
+        super(TestPvcMultiSnapshotPerformance, self).setup()
+
+        # Getting the total Storage capacity
+        try:
+            self.ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        except Exception as err:
+            err_msg = f"Failed to get Storage capacity : {err}"
+            log.error(err_msg)
+            raise Exception(err_msg)
+
+        # Use 70% of the storage capacity in the test
+        self.capacity_to_use = int(self.ceph_capacity * 0.7)
+
+        # Creating new namespace for the test
+        self.nss_name = "pas-test-namespace"
+        log.info(f"Creating new namespace ({self.nss_name}) for the test")
+        try:
+            self.proj = helpers.create_project(project_name=self.nss_name)
+        except CommandFailed as ex:
+            if str(ex).find("(AlreadyExists)"):
+                log.warning("The Namespace is Already Exists !")
```

Comment:
small spelling fix: 

"The namespace already exits! " instead of "The Namespace is Already Exists !"